reserved_ipv6.go

```diff
@@ -0,0 +1,139 @@
+package godo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const resourceV6Type = "ReservedIPv6"
+const reservedIPV6sBasePath = "v2/reserved_ipv6"
+
+// ReservedIPV6sService is an interface for interfacing with the reserved IPV6s
+// endpoints of the Digital Ocean API.
+type ReservedIPV6sService interface {
+	List(context.Context, *ListOptions) ([]ReservedIPV6, *Response, error)
+	Get(context.Context, string) (*ReservedIPV6, *Response, error)
+	Reserve(context.Context, *ReservedIPV6ReserveRequest) (*ReservedIPV6, *Response, error)
+	Release(context.Context, string) (*Response, error)
```

Comment:
wdyt about `Create` and `Delete` instead of `Reserve` and `Release`?

Reply:
umm, we can go with Create and Delete too ( thinking from a user POV), for v4 we have `Create` and `Delete` methods.
I went with Reserve and Release as we are using that terminology everywhere.

Reply:
Since this is customer facing and a charg-able resource, its better to name from their POV.